File: database/database-annotation-processor/src/main/java/io/koraframework/database/annotation/processor/cassandra/extension/CassandraTypesExtensionFactory.java
package io.koraframework.database.annotation.processor.cassandra.extension;

import io.koraframework.database.annotation.processor.cassandra.CassandraTypes;
import io.koraframework.kora.app.annotation.processor.extension.ExtensionFactory;
import io.koraframework.kora.app.annotation.processor.extension.KoraExtension;

import javax.annotation.processing.ProcessingEnvironment;
import java.util.Optional;

public class CassandraTypesExtensionFactory implements ExtensionFactory {
    @Override
    public Optional<KoraExtension> create(ProcessingEnvironment processingEnvironment) {
        var type = processingEnvironment.getElementUtils().getTypeElement(CassandraTypes.ROW_MAPPER.canonicalName());
        if (type == null) {
            return Optional.empty();
        }

        return Optional.of(new CassandraTypesExtension(processingEnvironment));
    }
}

File: database/database-symbol-processor/src/main/kotlin/io/koraframework/database/symbol/processor/cassandra/extension/CassandraTypesExtensionFactory.kt
package io.koraframework.database.symbol.processor.cassandra.extension

import com.google.devtools.ksp.processing.CodeGenerator
import com.google.devtools.ksp.processing.KSPLogger
import com.google.devtools.ksp.processing.Resolver
import io.koraframework.kora.app.ksp.extension.ExtensionFactory

class CassandraTypesExtensionFactory: ExtensionFactory {
    override fun create(resolver: Resolver, kspLogger: KSPLogger, codeGenerator: CodeGenerator) = CassandraTypesExtension(resolver, kspLogger, codeGenerator)
}
